### ISSUE TITLE
Spellcasting focus discoverability / focus attribution inconsistency

### ISSUE THREAD

**Issue by aaclayton** [foundryvtt/dnd5e#1399]
Originally in GitLab by @iaguastalli

TL;DR: its hard to find spellcasting focus, inconsistencies between focus types, classes and items. suggested action plan at the bottom.

While working on the Hero Creation Module, I noticed my players have a consistently hard time finding the corresponding spellcasting foci for their characters. Right now there's no clear way of finding a type of spellcasting focus programmatically, other than knowing the names beforehand, as far as I can tell.

But aside from the programmatic discoverability, it's also rather entangled for the players, and you need to know where to look for it, and the overall presentation of different focus types varies wildly. I'll give several examples:

In the first screenshot you will notice that:
1) Cleric class just mentions "Holy Symbol" without any link, reference or list.
2) Spellcasting feature for the class links to Adventuring Gear
3) Adventuring Gear lists examples of holy symbols (I understand the list is just for DM's inspiration in all likelihood, but at least linking to the actual items created on the SRD compendium would be nice) - Amulet and Emblem are at least found by the name, but the text "Reliquary" is nowhere to be found.
4) The items that mention being a Spellcasting Focus for Clerics & Paladins (Amulet, Emblem and Reliquary on the SRD items compendium) have no flag or reference to that attribution on the item details.
5) All that is to be found is a reference foot note about it on the item description.

![image](https://gitlab.com/foundrynet/dnd5e/uploads/42b5942dc109f47ab82f1f82be47d5ba/image.png)

-----------------------

In this case, for the Bard:
1) Class mentions "Lute or other musical instruments", again, with no link to any of those. (I understand this is probably done for consistency, as the weapons and armor are also not linked)
2) Spellcasting feature looks similar to Cleric at least, it point to Adventuring Gear.
3) .. but Adventuring Gear doesn't even have an entry for Musical Instruments! Neither it's on the list (2nd screenshot - notice no entry for either Lute on the L, nor Musical Instruments with the list as done for Holy Symbols and Druidic/Arcane foci); they are actually listed, correctly, on the `Tools` journal entry (3nd screenshot)
4) Item again doesn't refer to its attribution as a spellcasting focus (for bards, or overall), but it's also a `Tool` instead of an `Equipment - Tricket`. This will also be relevant for the Artificer eventually.
5) Description also varies; unlike the Cleric (and others), there's no bolded Spellcasting Focus subtitle, its just meandering on the description.

![image](https://gitlab.com/foundrynet/dnd5e/uploads/8c1f39c6414c6f98e4e965dc68e82ec8/image.png)
![image](https://gitlab.com/foundrynet/dnd5e/uploads/292c3b3f047e59a643314d9072f89c31/image.png)
![image](https://gitlab.com/foundrynet/dnd5e/uploads/cf2281ccda20d808e00da5e42be6d49e/image.png)

------------------------

For Arcane spellcasters / Druids, things get a little more streamlined:

1) Class mentions the foci generically, like the rest; no weirdness here.
2) But then, when looking at the Features, Druid one looks like the rest, with a link to Adventuring Gear, but Wizard has none (Pact magic and Paladin's are shown like Wizard's, without any link)
3) On the Adventuring Gear journal entry, at least, both Arcane Foci and Druidic Foci are listed pretty extensively; sure some things like the Yew Wand need a little thought, but at least they are both in the descriptions, and then listed individually (2nd screenshot)
4) In this case, these foci are listed as `Weapon`, and have a Focus flag, which I think its a better approach (though it makes no distinction as to whether its an arcane, druidic, divine or bardic focus).
5) As per descriptions, they are consistent with the Holy Symbol ones (IMO, the best presentation)

![image](https://gitlab.com/foundrynet/dnd5e/uploads/f1c531a9e658b0d33806e2c2746c15a6/image.png)
![image](https://gitlab.com/foundrynet/dnd5e/uploads/45a4dc91432465df72797fbc998faa67/image.png)

------------------------------------

From this, I wanted to show that it's both non intuitive to find by hand, by a player, and also hard programmatically to find appropriate foci (in my use case, I would like to present players with the option to select an appropriate focus when picking starting gear, or for them to purchase/select starting items, and right now it's impossible for them to find say an Arcane Focus by that name; they need to know they are searching for an Orb or a Wand or whatever. 

I have no way of linking the string "arcane focus" on a searchbox like this one to the individual options, other than specifically linking item ids X, Y and Z -or item names A, B and C- to contain the Arcane Focus or be found by the arcane focus search string. (hardcoded or tied to a module setting or something)

e.g screenshot from the Hero Creation Tool
![image](https://gitlab.com/foundrynet/dnd5e/uploads/6e9d70f1f34250e8d7a4ab824768e94a/image.png)![image](https://gitlab.com/foundrynet/dnd5e/uploads/9a2c18ed88469cb77ba5f4b97d5585a0/image.png)

Suggestion
---

From this, I think there are three points to address:

1) Standardize how spellcasting are shown to the user just clicking stuff and following links:
- all items that can be a focus should have a little subtitle in the description with the appropriate text at the bottom.
- all spellcasting features should link to the journal entry that lists standard options (`Adventuring Gear` for all except for Bard, that should link to the `Tools` journal entry)
- as an alternative to the last point, maybe list all the examples directly on the Spellcasting Focus subtitle on the feature (e.g. "You can use a holy symbol (like an Amulet, an Emblem or a Reliquary) as a spellcasting focus for your cleric spells." - all with the appropriate links to the SRD Item)

_(for this point I initially considered requesting adding the focus type to the name, so something like `Orb` would become `Orb (Arcane Focus)`, but I was rightfully refuted that wouldn't make sense, as  we don't now list weapons or armor by type either)_

2) Define a link between spellcasting and the supported focus:
- Even if its mentioned on the Spellcasting feature, I think its better suited for the Class item, where we already have the spellcasting `ability` and `progression` under the `data.spellcasting` key.
- How and what to link is another problem. For bards I imagine an approach bound to Tool types was handled in https://gitlab.com/foundrynet/dnd5e/-/issues/1081 could work here, linking the class' spellcasting to a Musical Instrument; for other spellcasters, I think something else would need to be worked.

3) Define what focus type an item is:
- The `Focus` flag on weapons is a good approximation, but as it's not present on Tools nor other non-weapon items. Maybe bringing that flag up to Item5e instead, so any kind of item can be a focus?
- Least in important imo, but for completeness-sake I imagine instead of a single `Focus` flag, that could be split between `Arcane Focus`, `Druidid Focus`, `Divine Focus` and `Bardic Focus`. That way each focus type can have a pinpoint reference to search for, for example if you want to present the players with all Arcane Focus items available on a compendium, and any magic item that could maybe be used as a generic Spellcasting Focus can just check all boxes.

If we define an standard approach for point 1, I can try and make a MR of this right away, but I imagine points 2 and 3 require a more in-depth discussion.
